clients/pkg/promtail/targets/serverutils/config.go

```diff
@@ -0,0 +1,29 @@
+package serverutils
+
+import (
+	"flag"
+
+	"github.com/imdario/mergo"
+	"github.com/weaveworks/common/server"
+)
+
+// MergeWithDefaults applies server.Config defaults to a given and different server.Config.
+func MergeWithDefaults(config server.Config) (server.Config, error) {
```

Comment:
This functionality was shared across all promtails target spinning up an http server (loki push, heroku and this new gcp push one), so extracted it in a `serverutils` package. Happy to rename it or move it elsewhere.